Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -0,0 +1,240 @@
+            {
+                m_PositionAction.performed += OnPositionUpdate;
+                m_PositionBound = true;
+                m_PositionAction.Enable();
+            }
+        }
+
+        void BindRotation()
+        {
+            if (!m_RotationBound && m_RotationAction != null)
+            {
+                m_RotationAction.performed += OnRotationUpdate;
+                m_RotationBound = true;
+                m_RotationAction.Enable();
+            }
+        }
+
+        void UnbindActions()
+        {
+            UnbindPosition();
+            UnbindRotation();           
+        }
+
+        void UnbindPosition()
+        {
+            if (m_PositionAction != null && m_PositionBound)
+            {
+                m_PositionAction.Disable();
+                m_PositionAction.performed -= OnPositionUpdate;
+                m_PositionBound = false;
+            }
+        }
+
+        void UnbindRotation()
+        {
+            if (m_RotationAction != null && m_RotationBound)
+            {
+                m_RotationAction.Disable();
+                m_RotationAction.performed -= OnRotationUpdate;
+                m_RotationBound = false;
+            }
+        }
+
+        void OnPositionUpdate(InputAction.CallbackContext context)
+        {
+            Debug.Assert(m_PositionBound);
+            m_CurrentPosition = context.ReadValue<Vector3>();                
+        }
+
+        void OnRotationUpdate(InputAction.CallbackContext context)
+        {
+            Debug.Assert(m_RotationBound);           
+            m_CurrentRotation = context.ReadValue<Quaternion>();                           
+        }
+
+        protected virtual void Awake()
+        {
+           
+#if ENABLE_VR
+            if (HasStereoCamera())
+            {
+                XRDevice.DisableAutoXRCameraTracking(GetComponent<Camera>(), true);
+            }
+#endif
+        }
+
+        protected void OnEnable()
+        {
+            InputSystem.onAfterUpdate += UpdateCallback;
+            BindActions();
+        }
+
+        void OnDisable()
+        {
+            UnbindActions();
+            InputSystem.onAfterUpdate -= UpdateCallback;
+        }
+
+        protected virtual void OnDestroy()
+        {
+#if ENABLE_VR
+            if (HasStereoCamera())
+            {
+                XRDevice.DisableAutoXRCameraTracking(GetComponent<Camera>(), false);
+            }
+#endif
+        }
+
+        protected void UpdateCallback(InputUpdateType type)
+        {
+            switch(type)
+            {
+                case InputUpdateType.Fixed:
+                    FixedUpdate();
+                    break;
+                case InputUpdateType.Dynamic:
+                    OnUpdate();
+                    break;
+                case InputUpdateType.BeforeRender:
+                    OnBeforeRender();
+                    break;                
+            }
+        }
+
+        protected virtual void FixedUpdate()

```

Comment:
Do we really want to update in FixedUpdate?
At the moment it's no different than the Dynamic update data, and can run multiple times per frame, wasting time setting the same data repeatedly.

Reply:
removed